src/sentry/static/sentry/app/views/settings/organization/members/organizationMembersView.jsx

```diff
@@ -0,0 +1,270 @@
+import {browserHistory} from 'react-router';
+import React from 'react';
+
+import {t, tct} from '../../../../locale';
+import Button from '../../../../components/buttons/button';
+import ConfigStore from '../../../../stores/configStore';
+import IndicatorStore from '../../../../stores/indicatorStore';
+import SentryTypes from '../../../../proptypes';
+import SpreadLayout from '../../../../components/spreadLayout';
+import OrganizationSettingsView from '../../../organizationSettingsView';
+import OrganizationMemberRow from './organizationMemberRow';
+import OrganizationAccessRequests from './organizationAccessRequests';
+
+class OrganizationMembersView extends OrganizationSettingsView {
+  static contextTypes = {
+    organization: SentryTypes.Organization
+  };
+
+  // XXX(billy): setState causes re-render of the entire view...
+  // we should not do this
+  getDefaultState() {
+    let state = super.getDefaultState();
+    return {
+      ...state,
+      members: [],
+      invited: new Map(),
+      accessRequestBusy: new Map()
+    };
+  }
+
+  getEndpoints() {
+    return [
+      ['members', `/organizations/${this.props.params.orgId}/members/`],
+      ['authProvider', `/organizations/${this.props.params.orgId}/auth-provider/`],
+      ['requestList', `/organizations/${this.props.params.orgId}/access-requests/`]
+    ];
+  }
+
+  getTitle() {
+    let org = this.context.organization;
+    return `${org.name} Members`;
+  }
+
+  removeMember = id => {
+    let {params} = this.props;
+    let {orgId} = params || {};
+
+    return new Promise((resolve, reject) => {
+      this.api.request(`/organizations/${orgId}/members/${id}/`, {
+        method: 'DELETE',
+        data: {},
+        success: data => {
+          this.setState(state => ({
+            members: state.members.filter(({id: existingId}) => existingId !== id)
```

Comment:
I prefer the destructured version because you're able to see the properties that you really care about - however I think you have a point that we don't really use restructuring anywhere else so it might not read as cleanly for others.